src/Source/DoctrineODMQuerySourceIterator.php

```diff
@@ -64,7 +70,23 @@ public function __construct(Query $query, array $fields, $dateTimeFormat = 'r')
             }
         }
 
-        $this->dateTimeFormat = $dateTimeFormat;
+        $formatters = array(new ArrayFormatter(), new DateTimeFormatter($dateTimeFormat), new ObjectToStringFormatter());
```

Comment:
Because it's a DIC violation. Dependencies should be injected. It's also an OCP violation. If someday we deem a new formatter is worth adding by default, we should't edit this file, we should just create a new service.